### ISSUE TITLE
IonInput autocomplete does nothing on iOS

### ISSUE THREAD

**Issue by scottgeye** [ionic-team/ionic-framework#22232]
#Bug Report

**Ionic version:**

[ ] **4.x**
[x] **5.x**

**Current behavior:**

When I use an IonInput component in React and set the autocomplete property to any value, no autocomplete values show up when I run on an iOS device and tap on the input.

**Expected behavior:**

Tapping on the input should properly show autocomplete values based on the what the autocomplete property was set to.

**Steps to reproduce:**

I created a new conference ionic starter app with React and added an autocomplete="on" prop to the username field of the login page.  I then ran the app on an iOS device and tapped the input.

**Comment by liamdebeasi** [comment]
Thanks for the issue. Is "Predictive" enabled on your iOS device? You can check in Settings > General > Keyboard > Predictive. Without that setting enabled, no autocomplete options will show up even if you do `autocomplete="on"`.

**Comment by scottgeye** [comment]
Yes, Predictive is on.

**Comment by liamdebeasi** [comment]
Are you expecting autocomplete options to show up when you tap into an empty input? iOS only shows options when the input has a value.

**Comment by scottgeye** [comment]
Here's what I'm seeing with our without anything typed into the input field:

![Screen Shot 2020-10-01 at 3 58 22 PM](https://user-images.githubusercontent.com/651989/94871532-0a57c480-03ff-11eb-9431-abfa3a0b997d.png)

I've also tried setting the autocomplete prop to "email" just in case it needed more specificity, but neither work.

**Comment by liamdebeasi** [comment]
Thanks! I think I figured out what is going on:

1. The "suggestion" bar you see on iOS with the 3 different options is actually controlled by the `autocorrect` property.
2. The `autocomplete` property lets you tweak how "specific" the autocorrect options are. For example, doing `autocomplete="name"` will show your name as an option in the bar.
3. Ionic Framework sets `autocorrect="off"` by default, which will disable this bar entirely.

In short, you need to set `autocorrect="on"` in order for this to work. I will make a note to update the docs for this since this is pretty confusing. The `ion-input` component relies on the native `input` element under the hood, so I tested this and you'd run into the same issue if you did `<input autocorrect="off" />`.

**Comment by scottgeye** [comment]
I was able to get this working on a text input field, but it doesn't seem to work with any other types.  I'm using inputs with types email and password for a login page and nothing comes up with autocorrect and autocomplete set.  This would seem counter intuitive given the autocomplete field allows for options like "email" and "new-password", but only works with text input.

**Comment by liamdebeasi** [comment]
Yes, the autocorrect and autocomplete properties can be a bit finicky. `ion-input` just uses the native `input` under the hood so we have control over the properties, but sometimes the OS puts restrictions around this. I'd recommend reading this article on autocomplete for more clarification: https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete

**Comment by scottgeye** [comment]
I'm wondering if this might be a webview specific issue if we can rule out ionic having any control over the functionality.  The docs you linked to don't indicate why email or password autocomplete wouldn't work.  I also made a test that only uses native input with autocomplete="email" and the correct autocomplete options show up when I open it in Safari on iOS:

http://plnkr.co/edit/22H0mk8Yl0gf9wTi

**Comment by scottgeye** [comment]
@liamdebeasi Any thoughts on how to proceed here?  The input component definitely doesn't behave the same way when run as an iOS app vs directly in browser.  Does that make it a capacitor issue?

**Comment by liamdebeasi** [comment]
Hmm yeah I have seen one or two issues that only apply in WKWebView. I'll check it out.

**Comment by liamdebeasi** [comment]
Oh actually I reported something similar to the WebKit team last year: https://bugs.webkit.org/show_bug.cgi?id=203299. Does this look like the issue you are running into?

**Comment by scottgeye** [comment]
Yeah, that looks like the same issue.  I'm thinking the ionic docs should be updated to indicate that only input of type text works with autocomplete until that bug is resolved.
